external/fv3fit/tests/training/test_cli.py

```diff
@@ -89,35 +82,118 @@ def test_local_download_path(
     ],
 )
 @pytest.mark.parametrize(
-    "validation_timesteps", [True, False],
+    "use_validation_timesteps", [True, False],
 )
 def test_training_integration(
-    model_info, data_info, validation_timesteps, tmp_path: str,
+    model_info, data_info, use_validation_timesteps: bool, tmp_path: str,
 ):
     """
     Test the bash endpoint for training the model produces the expected output files.
     """
-    config = ModelTrainingConfig(
-        data_path=data_info["data_path"],
+    (
+        training_filename,
+        train_data_filename,
+        validation_data_filename,
+        output_path,
+    ) = get_config(model_info, data_info, tmp_path, use_validation_timesteps)
+
+    subprocess.check_call(
+        [
+            "python",
+            "-m",
+            "fv3fit.train",
+            training_filename,
+            train_data_filename,
+            validation_data_filename,
+            output_path,
+        ]
+    )
+    fv3fit.load(output_path)
+
+
+def get_config(model_info, data_info, tmp_path, use_validation_timesteps: bool):
+    """
+    Initialize configuration files and get paths required to run fv3fit.train
+    """
+    training_config = fv3fit.TrainingConfig(
         model_type=model_info["model_type"],
-        hyperparameters=model_info["hyperparameters"],
         input_variables=["air_temperature", "specific_humidity"],
         output_variables=["dQ1", "dQ2"],
-        batch_function="batches_from_geodata",
-        batch_kwargs=data_info["batch_kwargs"],
-        scaler_type="standard",
-        scaler_kwargs={},
+        hyperparameters=fv3fit.get_hyperparameter_class(model_info["model_type"])(
+            **model_info["hyperparameters"]
+        ),
         additional_variables=[],
-        random_seed=0,
-        validation_timesteps=data_info["validation_timesteps"]
-        if validation_timesteps
-        else [],
     )
+    train_data_config = loaders.BatchesConfig(
+        data_path=data_info["data_path"],
```

Comment:
@mcgibbon and I discussed removing the special handling of the data_path since not all mappers are represented by one canonical data_path (e.g. the hybrid). `data_path` can be passed in `mapper_kwargs`.

This PR doesn't touch BatchesConfig so this change may be out of scope, we can address it in a follow-up PR or oepn an issue.

Reply:
I'll probably do this as a quick separate PR before merging this in. I think we'd like to minimize the number of times the yaml API will change, so this is the time to get it in.

Reply:
OK it does actually need to be in this PR, because of how the old config treats `data_path`.

Reply:
The hybrid mapper isn't actually incompatible with BatchesConfig is it? I was actually quite happy to see the vague typing of the old data path (either or a string, a list of strings, or a list of a single string) go away.

Reply:
It's not incompatible, and the vague typing is sort of gone. The difference now is that `data_path` is a kwarg like any other, instead of being its own special thing that must exist for every batches/mapper function. If we add configuration classes similar to the Hyperparameters classes for loaders, we can also restore strict typing.